crates/burn-import/onnx-tests/tests/constant_of_shape/constant_of_shape.py

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env python3
+
+# used to generate model: constant_of_shape.onnx
+
+# torch simplifies simple usecases where it can statically determine the shape of the constant
+# to use just ONNX constants instead of ConstantOfShape
+# Hence this model is exported using onnx directly
+
+import onnx
+import onnx.helper
+
+
+def build_model():
+    return onnx.helper.make_model(
+        ir_version=8,
+        opset_imports=[onnx.helper.make_operatorsetid("", 16)],
+        graph=onnx.helper.make_graph(name="main_graph", nodes=[
+            onnx.helper.make_node(
+                "ConstantOfShape",
+                inputs=["input1"],
+                outputs=["output1"],
+                name="/ConstantOfShape",
+                value=onnx.helper.make_tensor("value", data_type=onnx.TensorProto.FLOAT, dims=[1], vals=[1.125])
+            ),
+        ],
+        inputs=[
+            onnx.helper.make_value_info(
+                name="input1",
+                type_proto=onnx.helper.make_tensor_type_proto(
+                    elem_type=onnx.TensorProto.INT64, shape=[2, 3, 2]
```

Comment:
Is the onnx graph generated valid?

The input is supposed to be a 1D tensor with values `[2, 3, 2]`, not shape `[2, 3, 2]`.

You can check with `onnx.checker.check_model`

Reply:
Fixed it by introducing shape type.

Reply:
Thanks! I fixed it and actually it helped identify a bug in dim inference.